File: quickjs/test/trajpoly3.js
import * as a from "liba.so";
function assert(expr) {
    if (!eval(expr)) {
        throw Error(expr);
    }
}
var ctx = new a.trajpoly3(1, 0, 1);
ctx.gen(10, 0, 10, 0, 0);
assert("ctx.c0");
assert("ctx.c1");
assert("ctx.c2");
for (var i = 0; i < 10; i += 0.1) {
    console.log(i, ctx.pos(i), ctx.vel(i), ctx.acc(i));
}

File: tests/trajpoly3.rs
#[test]
fn trajpoly3() {
    let x = 0.5;
    {
        let traj = liba::trajpoly3::new(1.0, 0.0, 1.0, 0.0, 1.0);
        std::println!("[{}, {}, {}]", traj.pos(x), traj.vel(x), traj.acc(x));
    }
}
